pkg/platform/microservice/service.go

```diff
@@ -113,12 +108,99 @@ func (s *service) Create(responseWriter http.ResponseWriter, request *http.Reque
 	case platform.MicroserviceKindRawDataLogIngestor:
 		s.handleRawDataLogIngestor(responseWriter, request, requestBytes, applicationInfo)
 	case platform.MicroserviceKindPurchaseOrderAPI:
-		s.purchaseOrderHandler.Create(responseWriter, request, requestBytes, applicationInfo)
+		purchaseOrderAPI, err := s.purchaseOrderHandler.Create(requestBytes, applicationInfo)
+		if err != nil {
+			utils.RespondWithError(responseWriter, err.StatusCode, err.Error())
+		}
+		utils.RespondWithJSON(responseWriter, http.StatusAccepted, purchaseOrderAPI)
+	default:
+		utils.RespondWithError(responseWriter, http.StatusBadRequest, fmt.Sprintf("Kind %s is not supported", microserviceBase.Kind))
+	}
+}
+
+func (s *service) Update(responseWriter http.ResponseWriter, request *http.Request) {
+	logger := s.logger.WithFields(logrus.Fields{
+		"method": "Update",
+	})
+	requestBytes, microserviceBase, err := s.readMicroserviceBase(request, logger)
+	if err != nil {
+		utils.RespondWithError(responseWriter, http.StatusBadRequest, fmt.Errorf("Invalid request payload: %w", err).Error())
+		return
+	}
+	defer request.Body.Close()
+
+	applicationID := microserviceBase.Dolittle.ApplicationID
+	environment := microserviceBase.Environment
+
+	applicationInfo, err := s.k8sDolittleRepo.GetApplication(applicationID)
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			fmt.Println(err)
+			utils.RespondWithError(responseWriter, http.StatusInternalServerError, "Something went wrong")
+			return
+		}
+
+		utils.RespondWithJSON(responseWriter, http.StatusNotFound, map[string]string{
+			"error": fmt.Sprintf("Application %s not found", applicationID),
+		})
+		return
+	}
+
+	customer := k8s.Tenant{
+		ID:   applicationInfo.Tenant.ID,
+		Name: applicationInfo.Tenant.Name,
+	}
+
+	userID := request.Header.Get("User-ID")
+	allowed := s.k8sDolittleRepo.CanModifyApplicationWithResponse(responseWriter, customer.ID, applicationID, userID)
+	if !allowed {
+		return
+	}
+
+	if !s.gitRepo.IsAutomationEnabled(customer.ID, applicationID, environment) {
+		utils.RespondWithError(
+			responseWriter,
+			http.StatusBadRequest,
+			fmt.Sprintf(
+				"Customer %s with application %s in environment %s does not allow changes via Studio",
+				customer.ID,
+				applicationID,
+				environment,
+			),
+		)
+		return
+	}
+
+	switch microserviceBase.Kind {
+	case platform.MicroserviceKindPurchaseOrderAPI:
+		// This is not ideal, we need to
```

Comment:
```suggestion
		// TODO handle other updation operations too
```

Forgot to finish the sentence, here's my attempt at it

Reply:
Opps :D